### ISSUE TITLE
[Bug]: 'start' input is not a scalar

### ISSUE THREAD

**Issue by jikechao** [openvinotoolkit/openvino#21178]
### OpenVINO Version

openvino-nightly 2023.2.0.dev20231101

### Operating System

Ubuntu 18.04 (LTS)

### Device used for inference

CPU

### Framework

ONNX

### Model used

https://github.com/jikechao/onnx_models/blob/main/Range.onnx

### Issue description

![image](https://github.com/openvinotoolkit/openvino/assets/29506758/adf54201-e57e-4168-95c9-f27890a68570)

In ONNX, the attribute `start` of `Range` supports tensor with a single value (e.g., [1]) and can run well.
Although the ONNX requirement shows that the type of `start` is scalar, it can convert the tensor with a single value automatically to the scalar.
Thus, for support such models, OV should also support such case.

### Step-by-step reproduction

```
import openvino as ov

onnx_model_path = 'Range.onnx'
ov_model = ov.convert_model(onnx_model_path)
```

### Relevant log output

```shell
Traceback (most recent call last):
  File "test.py", line 4, in <module>
    ov_model = ov.convert_model(onnx_model_path)
  File "C:\software\conda\envs\torch\lib\site-packages\openvino\tools\ovc\convert.py", line 101, in convert_model
    ov_model, _ = _convert(cli_parser, params, True)
  File "C:\software\conda\envs\torch\lib\site-packages\openvino\tools\ovc\convert_impl.py", line 524, in _convert
    raise e
  File "C:\software\conda\envs\torch\lib\site-packages\openvino\tools\ovc\convert_impl.py", line 476, in _convert
    ov_model = driver(argv, {"conversion_parameters": non_default_params})
  File "C:\software\conda\envs\torch\lib\site-packages\openvino\tools\ovc\convert_impl.py", line 226, in driver
    ov_model = moc_emit_ir(prepare_ir(argv), argv)
  File "C:\software\conda\envs\torch\lib\site-packages\openvino\tools\ovc\convert_impl.py", line 172, in prepare_ir
    ov_model = moc_pipeline(argv, moc_front_end)
  File "C:\software\conda\envs\torch\lib\site-packages\openvino\tools\ovc\moc_frontend\pipeline.py", line 247, in moc_pipeline
    ov_model = moc_front_end.convert(input_model)
  File "C:\software\conda\envs\torch\lib\site-packages\openvino\frontend\frontend.py", line 18, in convert
    converted_model = super().convert(model)
RuntimeError: Check 'error_message.empty()' failed at src\frontends\onnx\frontend\src\frontend.cpp:124:
Errors during ONNX translation: 
While validating ONNX node '<Node(Range): output>':
Check 'input_shapes[0].rank().compatible(0)' failed at src\core\shape_inference\include\range_shape_inference.hpp:24:
While validating node 'opset4::Range Range_4 (start[0]:f32[1], limit[0]:f32[1], delta[0]:f32[1]) -> (dynamic[...])' with friendly_name 'Range_4':
'start' input is not a scalar
```

### Issue submission checklist

- [X] I'm reporting an issue. It's not a question.
- [X] I checked the problem with the documentation, FAQ, open issues, Stack Overflow, etc., and have not found a solution.
- [X] There is reproducer code and related data files such as images, videos, models, etc.

**Comment by siddhant-0707** [comment]
Hey @jikechao the issue is resolved! Do contact if you face any other issues.
PR #21741

**Comment by jikechao** [comment]
@siddhant-0707 Thanks for your fixing. I closed this issue due to the fixing in #21741.
